paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentElementLoader.kt

```diff
@@ -494,14 +477,31 @@ internal class DefaultPaymentElementLoader @Inject constructor(
         configuration: CommonConfiguration,
         customer: CustomerInfo?,
         elementsSession: ElementsSession,
-        merchantCountry: String?,
-        passthroughModeEnabled: Boolean,
-        flags: Map<String, Boolean>,
-        useAttestationEndpointsForLink: Boolean,
-        suppress2faModal: Boolean,
         initializationMode: PaymentElementLoader.InitializationMode
-    ): LinkConfiguration {
-        val shippingDetails: AddressDetails? = configuration.shippingDetails
+    ): LinkConfiguration? {
+        if (!configuration.link.shouldDisplay ||
+            configuration.billingDetailsCollectionConfiguration.collectsAnything ||
+            !elementsSession.isLinkEnabled
+        ) {
+            return null
+        }
+
+        val isCardBrandFilteringRequired =
+            elementsSession.linkPassthroughModeEnabled &&
+                configuration.cardBrandAcceptance != PaymentSheet.CardBrandAcceptance.All
+
+        if (isCardBrandFilteringRequired && !FeatureFlags.linkCardBrandFiltering.isEnabled) {
+            return null
+        }
```

Comment:
We need one more condition here: Is the merchant using native Link or the web flow?

We can't support CBF in the web flow right now, so we'd need to turn off Link web in those cases.

Reply:
Updated: https://github.com/stripe/stripe-android/pull/10759/commits/905909f1b0baf0c0c8a4636f716c7b7e4fedba32